internal/gengapic/testdata/deprecated_client_init.want

```diff
@@ -133,16 +133,56 @@ func (c *gRPCClient) Close() error {
 
 // Methods, except Close, may be called concurrently. However, fields must not be modified concurrently with method calls.
 type restClient struct {
-	host string
+	// The http endpoint to connect to.
+	endpoint string
+
+	// The http client.
+	httpClient *http.Client
+
+	// The x-goog-* metadata to be sent with each request.
+	xGoogMetadata metadata.MD
 }
 
-// Zip does some stuff.
-func (c *restClient) Zip(ctx context.Context, req *mypackagepb.Bar, opts ...gax.CallOption) (*mypackagepb.Foo, error) {
-	m := jsonpb.Marshaler{}
-	if jsonReq, err := m.MarshalToString(req); err != nil {
+// NewRESTClient creates a new foo rest client.
+//
+// Foo service does stuff.
+//
+// Deprecated: Foo may be removed in a future version.
+func NewRESTClient(ctx context.Context, opts ...option.ClientOption) (*Client, error) {
+	clientOpts := append(defaultRESTClientOptions(), opts...)
+	httpClient, endpoint, err := httptransport.NewClient(ctx, clientOpts...)
+	if err != nil {
 		return nil, err
+	}
 
+	c := &restClient{
+		endpoint: endpoint,
+		httpClient: httpClient,
 	}
-	return nil, nil
+	c.setGoogleClientInfo()
+
+	return &Client{internalClient: c, CallOptions: defaultCallOptions()}, nil
+}
 
+// setGoogleClientInfo sets the name and version of the application in
+// the `x-goog-api-client` header passed on each request. Intended for
+// use by Google-written clients.
+func (c *restClient) setGoogleClientInfo(keyval ...string) {
+	kv := append([]string{"gl-go", versionGo()}, keyval...)
+	kv = append(kv, "gapic", versionClient, "gax", gax.Version, "rest", "UNKNOWN")
+	c.xGoogMetadata = metadata.Pairs("x-goog-api-client", gax.XGoogHeader(kv...))
+}
+
+// Close closes the connection to the API service. The user should invoke this when
+// the client is no longer required.
+func (c *restClient) Close() error {
+	c.httpClient.CloseIdleConnections()
+	return nil
+}
+
+// Connection returns a connection to the API service.
```

Comment:
This can be done post-merge: I think we should comment on the HTTP version of this method that this doesn't work and just returns nil :) for completeness-sake

Reply:
Good call.